video/out/vo_libmpv.c

```diff
@@ -281,9 +284,10 @@ void mpv_render_context_free(mpv_render_context *ctx)
 
             // Wait for kill_cb() or update() calls.
             pthread_mutex_lock(&ctx->update_lock);
-            if (!ctx->had_kill_update)
+            if (!ctx->had_kill || !ctx->had_update)
                 pthread_cond_wait(&ctx->update_cond, &ctx->update_lock);
-            ctx->had_kill_update = false;
+            had_kill = ctx->had_kill;
+            ctx->had_update = false;
             pthread_mutex_unlock(&ctx->update_lock);
```

Comment:
a corner case where `had_kill == true && had_update == false` causes waiting infinitely, because there is no more `update` to wake it up when `had_kill == true`. try adding `mp_sleep_us(1000000);` after this line to reproduce the issue. removing `had_update` can fix it.